tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -447,16 +445,17 @@ def capture_route(self, request):
         return http_util.Respond(request,
             {'error': 'no TPUs with the specified names exist.'},
              'application/json', code=200)
-      workers_list = get_workers_list(tpu_cluster_resolver)
+      if not worker_list:
+        worker_list = get_worker_list(tpu_cluster_resolver)
       # TPU cluster resolver always returns port 8470. Replace it with 8466
       # on which profiler service is running.
       master_ip = master_grpc_addr.replace('grpc://', '').replace(':8470', '')
       service_addr = master_ip + ':8466'
       # Set the master TPU for streaming trace viewer.
       self.master_tpu_unsecure_channel = master_ip
     try:
-      profiler_client.start_tracing(service_addr, self.logdir,
-                                    duration, workers_list);
+      profiler_client.start_tracing(service_addr, self.logdir, duration,
+          worker_list, include_dataset_ops, num_tracing_attempts);
```

Comment:
nit: Superfluous semicolon on this line.

Reply:
Done, thanks for spotting this.